File: MB-Chat/imaging_py/provider_contract.py
from __future__ import annotations

from abc import ABC, abstractmethod

from .types import ImageAnalysisResult, ImageInput, ImageModality


class ImageProvider(ABC):
    provider_name: str

    @abstractmethod
    def analyze(self, input_data: ImageInput) -> ImageAnalysisResult:
        raise NotImplementedError

    @abstractmethod
    def healthcheck(self) -> dict[str, object]:
        raise NotImplementedError

    @abstractmethod
    def supports_modality(self, modality: ImageModality) -> bool:
        raise NotImplementedError


IMAGE_PROVIDER_CONTRACT_STATUS = {
    "implemented_providers": [],
    "enabled_by_default": False,
    "external_calls_allowed_in_phase_4": False,
}

File: MB-Chat/imaging/provider.contract.ts
import type { ImageAnalysisResult, ImageInput, ImageModality, ImageProviderHealth } from "./types";

export type ImageProvider = {
  readonly provider_name: string;
  analyze(input: ImageInput): Promise<ImageAnalysisResult>;
  healthcheck(): Promise<ImageProviderHealth>;
  supports_modality(modality: ImageModality): boolean;
};

export const IMAGE_PROVIDER_CONTRACT_STATUS = {
  implementedProviders: [],
  enabledByDefault: false,
  externalCallsAllowedInPhase4: false,
  note: "Contract only. Phase 4 does not send images to external providers.",
} as const;
